hcl2template/plugin.go

```diff
@@ -0,0 +1,101 @@
+package hcl2template
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"log"
+	"runtime"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/hashicorp/packer/packer"
+	plugingetter "github.com/hashicorp/packer/packer/plugin-getter"
+)
+
+// PluginRequirements returns a sorted list of plugin requirements.
+func (cfg *PackerConfig) PluginRequirements() (plugingetter.Requirements, hcl.Diagnostics) {
+
+	var diags hcl.Diagnostics
+	var reqs plugingetter.Requirements
+	reqPluginsBlocks := cfg.Packer.RequiredPlugins
+
+	// Take all required plugins, make sure there are no conflicting blocks
+	// and append them to the list.
+	uniq := map[string]*RequiredPlugin{}
+	for _, requiredPluginsBlock := range reqPluginsBlocks {
+		for name, block := range requiredPluginsBlock.RequiredPlugins {
+
+			if previouslySeenBlock, found := uniq[name]; found {
+				diags = append(diags, &hcl.Diagnostic{
+					Severity: hcl.DiagError,
+					Summary:  fmt.Sprintf("Duplicate required_plugin.%q block", name),
```

Comment:
👍 good explanation

Reply:
we're gonna want tests for all this logic

Reply:
Thanks ! Yes Totally ! Leaving this unresolved as a reminder 🙂